### PR TITLE
Fix/succesor markets

### PR DESCRIPTION
**Pull request by davidsiska-vega**
Few small updates on successor markets.

### CODE REVIEW

File: protocol/0028-GOVE-governance.md
Comment (EVODelavega) [inline protocol/0028-GOVE-governance.md]: So specifically: if a market has been accepted, enacted proposal still in opening auction?

File: protocol/0028-GOVE-governance.md
Comment (EVODelavega) [inline protocol/0028-GOVE-governance.md]: parent markets don't have a successor field. Successors have a parent market field. Is there a reason why this should be a doubly linked list?

File: protocol/0028-GOVE-governance.md
Comment (davidsiska-vega) [reply protocol/0028-GOVE-governance.md]: Yes. Clarified, will push.

File: protocol/0028-GOVE-governance.md
Comment (davidsiska-vega) [reply protocol/0028-GOVE-governance.md]: I assumed it will be, yes. I want to have a look at a market and see immediately whether I can use it as my parent or not.

Comment (EVODelavega) [reply protocol/0028-GOVE-governance.md]: From data-node, or does core have to provide this?

Comment (davidsiska-vega) [reply protocol/0028-GOVE-governance.md]: When a new proposal comes saying marketId `xyz` is the proposed parent core needs to do validation. 

With a single linked lists I will need to look through all the markets to see if any of them use `xyz` as parent. With a doubly linked list I just go to `xyz` and check whether its available. 

To me that sounds saner (though of course there is maybe more work to ensure consistency). 

I'll leave it up to your implementation @EVODelavega and you can come back and edit this part to match implementation. 

Data node will have to provide the double linked info because hypothetically with some obscure data retention rules the successor may be gone already from the date node so the single linked version won't do, I think.

Comment (EVODelavega) [reply protocol/0028-GOVE-governance.md]: I agree that datanode has to offer the capability to traverse the _"family tree"_/lineage of markets, and thus requires something like a doubly linked list, and datanode will therefore be able to provide you the information required (as in: market xyz has proposal abc as a successor).

As far as the core is concerned though, markets are pretty isolated things, they aren't really aware, nor do they have access to other markets that may exist. The execution engine, which instantiates the markets is aware of which market is the parent/successor, and once a successor market was enacted, all other proposals will get rejected as per spec (several competing successor proposals can exist after all). Once a successor proposal is enacted, any future proposal to succeed its parent will just get rejected outright, and by that time, the datanode can already provide the relevant information easily.

To ensure market state doesn't depend on the state of other markets once they are enacted, I think it best that core only concerns itself with setting the parent market id field on the market. Once that market is enacted, that field is either blank or not, and becomes static.

**Review by EVODelavega — DISMISSED** [review]

**Review by EVODelavega — DISMISSED** [review]

**Review by gordsport — APPROVED** [review]
reapproving as the only change after the approval from @EVODelavega was a typo (no content change)

**Pull request merged**
